retry/regular.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package retry
+
+import (
+	"time"
+
+	"github.com/juju/utils/clock"
+)
+
+// Regular represents a strategy that repeats at regular intervals.
+type Regular struct {
+	Total time.Duration // total duration of attempt.
+	Delay time.Duration // interval between each try in the burst.
+	Min   int           // minimum number of retries; overrides Total
```

Comment:
> But that's a completely different mechanism than the minimum, which is why it's confusing.

Yes, it's a different mechanism because you can't do the min count thing by strategy composition (what timing would the extra events get?). We could use MinCount too, but I honestly don't think that name is confusing as it's in the context of the Regular struct type and documented.

I've updated the field documentation for Regular to use a more conventional comment style
and to mention LimitCount.